content/blog/integrating-android-code-in-unity/index.md

```diff
@@ -0,0 +1,326 @@
+
+```groovy
+dependencies {
+	implementation fileTree(dir: '../../Assets/Plugins/Android', include: ['*.jar', '*.aar'])
+}
+```
+
+This will take all of the AAR files and JAR files and treat them as if they were synced by Android Studio's gradle sync
+
+
+
+For more information on how to manage your app's dependencies from within Unity, you may want to checkout [this article created by the Firebase developers](https://medium.com/firebase-developers/how-to-manage-your-native-ios-and-android-dependencies-in-unity-like-firebase-921659843aef), who coincidentally made the plugin for managing Android dependencies in Unity.
+
+
+
+# Call Android code from C# {#call-android-from-c-sharp}
+
+It's great that we're able to manage those dependencies, but they don't mean much if you're not able to utilize the code from them!
+
+For example, take the following library: https://github.com/jaredrummler/AndroidDeviceNames
+
+That library allows you to grab metadata about a user's device. This might be useful for analytics or bug reporters you may be developing yourself. Let's see how we're able to integrate this Java library in our C# code when building for the Android platform
+
+## Introduction {#intro-call-android-from-c-sharp}
+
+You must make your callback extend the type of callback that is used in the library. For example, take the following code sample from the README of the aforementioned library:
+
+```java
+DeviceName.with(context).request(new DeviceName.Callback() {
+	@Override public void onFinished(DeviceName.DeviceInfo info, Exception error) {
+		String manufacturer = info.manufacturer;  // "Samsung"
+		String name = info.marketName;            // "Galaxy S8+"
+		String model = info.model;                // "SM-G955W"
+		String codename = info.codename;          // "dream2qltecan"
+		String deviceName = info.getName();       // "Galaxy S8+"
+	}
+});
+```
+
+While this example may seem straightforward, let's disect what we're doing step-by-step here. This will allow us to make the migration to C# code much simpler to do mentally.
+
+```java
+// Create a new "DeviceName.Callback" instance
+DeviceName.Callback handleOnFinished = new DeviceName.Callback() {
+	// Provide an implementation of the `onFinished` function in the `Callback` class
+	// Notice that there are two parameters for this method: one for info, the other for errors
+	@Override public void onFinished(DeviceName.DeviceInfo info, Exception error) {
+		// ... Assignment logic here
+	}
+};
+
+// Create a `DeviceName.Request` by passing the current context into the `DeviceName.with` method
+DeviceName.Request withInstance = DeviceName.with(context);
+
+// Use that request instance to pass the `DeviceName.Callback` instance from above to run the related code
+withInstance.request(handleOnFinished);
+```
+
+You can see that we have a few steps here:
+
+1) Make a new `Callback` instance
+1a) Provide an implementation of `onFinished` for said instance
```

Comment:
```suggestion
	1) Provide an implementation of `onFinished` for said instance
```

I'd like to see these list items (also below) pulled out of the top level. It's hard to read and not the most accessible.

It may require some code or styling changes if you want letters for the second list, and also this bug affects us: https://github.com/gatsbyjs/gatsby/issues/10870

Reply:
Let's make a new bug for gatsbyjs issue and mark it an upstream problem